avm/ptn/app/container-job-toolkit/modules/deploy_services.bicep

```
@@ -0,0 +1,930 @@
+@description('Optional. The location for the resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. The name of the resource group for the resources.')
+param resourceGroupName string = resourceGroup().name
+
+@description('Required. The name will be used as suffix for newly created resources.')
+param name string
+
+// network related parameters
+// -------------------------
+@description('Optional. Deploy resources in a virtual network and use it for private endpoints.')
+param deployInVnet bool = true
+
+@description('Conditional. A new private DNS Zone will be created. Optional, if `deployInVnet` is `true`.')
+param deployDnsZoneKeyVault bool = true
+
+@description('Conditional. A new private DNS Zone will be created. Optional, if `deployInVnet` is `true`.')
+param deployDnsZoneContainerRegistry bool = true
+
+@description('Conditional. The address prefix for the virtual network needs to be at least a /16. Three subnets will be created (the first /24 will be used for private endpoints, the second /24 for service endpoints and the second /23 is used for the workload). Required, if `deployInVnet` is `true`.')
+param addressPrefix string
+
+@description('Conditional. CIDR notation IP range assigned to the Docker bridge, network. It must not overlap with any other provided IP ranges and can only be used when the environment is deployed into a virtual network. If not provided, it will be set with a default value by the platform. Required if `zoneRedundant` for consumption plan is desired or `deployInVnet` is `true`.')
+param dockerBridgeCidr string = '172.16.0.1/28'
+
+@description('Conditional. IP range in CIDR notation that can be reserved for environment infrastructure IP addresses. It must not overlap with any other provided IP ranges and can only be used when the environment is deployed into a virtual network. If not provided, it will be set with a default value by the platform. Required if `zoneRedundant` for consumption plan is desired or `deployInVnet` is `true`.')
+param platformReservedCidr string = '172.17.17.0/24'
+
+@description('Conditional. An IP address from the IP range defined by "platformReservedCidr" that will be reserved for the internal DNS server. It must not be the first address in the range and can only be used when the environment is deployed into a virtual network. If not provided, it will be set with a default value by the platform. Required if `zoneRedundant` for consumption plan is desired or `deployInVnet` is `true`.')
+param platformReservedDnsIP string = '172.17.17.17'
+
+@description('Optional. Network security group, that will be added to the workload subnet.')
+param customNetworkSecurityGroups securityRulesType
```

Please keep UDT singular and not nullable
```suggestion
param customNetworkSecurityGroups securityRuleType[]?
```